core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityManager.scala

```
@@ -60,14 +60,21 @@ class EntityManager(providerBuilders: Set[EntityProviderBuilder[_ <: EntityProvi
       )
     }
 
-    // If the workspace has the CompactDataTables setting enabled, use CompactEntityProvider; else use LocalEntityProvider.
-    val compactDataTables =
-      workspaceSettingRepository.getWorkspaceSettingOfType(requestArguments.workspace.workspaceIdAsUUID,
-                                                           CompactDataTables
-      ) map {
-        case Some(qs: CompactDataTablesSetting) => qs.config.enabled
-        case _                                  => false
-      }
+    // If the workspace has the CompactDataTables setting enabled and no pending CompactDataTables Settings,
+    // use CompactEntityProvider; else use LocalEntityProvider.
+    val compactDataTables = for {
+      settingOpt <- workspaceSettingRepository.getWorkspaceSettingOfType(
+        requestArguments.workspace.workspaceIdAsUUID,
+        CompactDataTables
+      )
+      hasPending <- workspaceSettingRepository.hasPendingSettings(
+        requestArguments.workspace.workspaceIdAsUUID,
+        CompactDataTables
+      )
+    } yield settingOpt match {
+      case Some(qs: CompactDataTablesSetting) if qs.config.enabled && !hasPending => true
+      case _                                                                      => false
+    }
```

I think this should throw an error if a local->compact migration is in progress. As implemented, this would still allow users to (attempt to) write to the local data tables in the middle of a migration. The db transactions should handle that but it could lead users to think their changes have been written, but those changes will not be reflected in the Quicksilver data tables.

We could get wording from Adam but I think an error explaining that data tables are mid-migration and no access is allowed would be better.

---

I have added the error message: https://github.com/broadinstitute/rawls/pull/3407/commits/b28628a500ae2da599ca16bafd161f66a9a2feaa